src/core/sensorlistmodel.cpp

```diff
@@ -52,16 +52,20 @@ void SensorListModel::setProject( QgsProject *project )
     disconnect( mProject->sensorManager(), &QgsSensorManager::sensorErrorOccurred, this, &SensorListModel::handleSensorError );
   }
 
+  if ( mSensorModel )
+  {
+    mSensorModel->deleteLater();
+    mSensorModel = nullptr;
+  }
+
   mProject = project;
   emit projectChanged();
 
-  connect( mProject->sensorManager(), &QgsSensorManager::sensorErrorOccurred, this, &SensorListModel::handleSensorError );
-
-  if ( mSensorModel )
+  if ( mProject )
   {
-    mSensorModel->deleteLater();
+    connect( mProject->sensorManager(), &QgsSensorManager::sensorErrorOccurred, this, &SensorListModel::handleSensorError );
+    mSensorModel = new QgsSensorModel( mProject->sensorManager() );
```

Comment:
Should `mSensorModel` be a `QObjectUniquePtr`?

Reply:
@m-kuhn , done.